rflx/pyrflx/typevalue.py

```diff
@@ -463,161 +535,153 @@ def accepted_type(self) -> type:
     def literals(self) -> Mapping[Name, Expr]:
         return {}
 
-    def assign(self, value: bytes, length: int = 0, check: bool = True) -> None:
+    @property
+    def size(self) -> Number:
+        return Number(len(self.bitstring))
+
+    def assign(self, value: bytes, check: bool = True) -> None:
 
         msg_as_bitstr: Bitstring = Bitstring().from_bytes(value)
-        self.assign_bitvalue(msg_as_bitstr, length)
+        self.parse(msg_as_bitstr)
 
-    def assign_bitvalue(self, value: Bitstring, length: int = 0) -> None:
-        """
-        :param value: Bitstring representation of the message
-        :param length: use length parameter if complete
-        length of the message is less than 1 Byte
-        """
+    def parse(self, value: Bitstring) -> None:
```

Comment:
1. How should the code behave when an incomplete message was found?
2. Why is no exception raised when there are not enough bits for the second field?

```Python
>>> frame = PyRFLX(["specs/ethernet.rflx"])["Ethernet"]["Frame"]
>>> frame.parse(Bitstring("0" * 50))
>>> frame.parse(Bitstring("0"))
Traceback (most recent call last):
  File "/home/tr/tmp/RecordFlux/rflx/pyrflx/typevalue.py", line 627, in parse
    ) = set_field_with_length(current_field_name, current_field_length)
  File "/home/tr/tmp/RecordFlux/rflx/pyrflx/typevalue.py", line 586, in set_field_with_length
    value[current_pos_in_bitstring : current_pos_in_bitstring + field_length],
  File "/home/tr/tmp/RecordFlux/rflx/pyrflx/bitstring.py", line 19, in __getitem__
    raise IndexError
IndexError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/tr/tmp/RecordFlux/rflx/pyrflx/typevalue.py", line 629, in parse
    raise IndexError(
IndexError: Bitstring representing the message is too short - stopped while parsing field: Destination
```

Reply:
Fixed by 6ac8336